File: backend/TaskService/Models/TaskModel.cs
namespace TaskService.Models
{

    public class TaskModel
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public TodoTaskStatus Status { get; set; }
        public TaskPriority Priority { get; set; }
        public int UserId { get; set; }
    }
}

File: frontend/app/src/app/core/models/task.model.ts
export type TaskStatus = 'Aberto' | 'EmAndamento' | 'Concluido';
export type TaskPriority = 'Baixa' | 'Media' | 'Alta';

export interface Task {
  id: number;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  createdAt: string;
  completionDate?: string;
  userId: number;
}

export interface CreateTaskDto {
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  completionDate?: string;
} 